### PR TITLE
First Submmit

### PR DESCRIPTION
**Pull request by OriVered**

### CODE REVIEW

**Comment by OriVered** [comment]
@excellenteam-scaleup

File: 5.1.thats_the_way.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+    Args:
+        path (str): The path to search for files starting with 'deep'.
+
+    Returns:
+        deep_files (list): A list of filenames in the given path that start with 'deep'.
+    """
+    deep_files = []
+    for filename in os.listdir(path):
+        if filename.startswith("deep"):
```
Comment (excellenteam-scaleup) [inline 5.1.thats_the_way.py:16]: This is known as a "magic string"
in general it is better to separate IO from logic in functions, but even if you want this function to be specific, save "deep" as a variable

you can read more about it here:  
https://wiki.c2.com/?SeparateIoFromCalculation

File: 5.1.thats_the_way.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+            deep_files.append(filename)
+    return deep_files
+
+
+if __name__ == "__main__":
```
Comment (excellenteam-scaleup) [inline 5.1.thats_the_way.py:21]: Nice! important check
but...
We will always call "main" function here and put the logic over there.
Read more: https://www.geeksforgeeks.org/python-main-function/amp/

File: 5.1.thats_the_way.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    path = "Enter_your_path"
```
Comment (excellenteam-scaleup) [inline 5.1.thats_the_way.py:22]: don't you forget something?

File: 5.1.i_dont_have_vinaigrette.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import random
+from datetime import datetime, timedelta
+
+def random_date_between(start_date, end_date):
+    """
+    Generate a random date between two given dates.
+
+    Args:
+        start_date (datetime): The start date.
```
Comment (excellenteam-scaleup) [inline 5.1.i_dont_have_vinaigrette.py:9]: Great

File: 5.1.i_dont_have_vinaigrette.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    end_date_str = input("Enter the second date (YYYY-MM-DD): ")
+
+    start_date = datetime.strptime(start_date_str, "%Y-%m-%d")
+    end_date = datetime.strptime(end_date_str, "%Y-%m-%d")
+
+    random_date = random_date_between(start_date, end_date)
+
+    print(f"Random date: {random_date.strftime('%Y-%m-%d')}")
+
+    if random_date.weekday() == 0:
```
Comment (excellenteam-scaleup) [inline 5.1.i_dont_have_vinaigrette.py:35]: good job

File: 5.2.a_peice_of_cake.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    Returns:
+        float: The total price of the recipe.
+    """
+    if optionals is None:
+        optionals = []
+
+    total_price = 0
+    for ingredient, quantity in ingredients.items():
+        if ingredient not in optionals:
+            price_per_100g = prices.get(ingredient, 0)
```
Comment (excellenteam-scaleup) [inline 5.2.a_peice_of_cake.py:19]: Nice use of dict.get

File: 5.2.a_peice_of_cake.py [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+            total_price += (quantity / 100) * price_per_100g
+
+    return total_price
+
+if __name__ == "__main__":
```
Comment (excellenteam-scaleup) [inline 5.2.a_peice_of_cake.py:24]: Nice testing.
now that you learned pytest, you can make it even more professional :)

File: 5.2.cup_of_join.py [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    Args:
+        *lists: A variable number of lists to join.
+        sep (str, optional): The separator to insert between lists. Defaults to '-'.
+
+    Returns:
+        list: A single list composed of all input lists with the separator inserted between them.
+        None: If no lists are provided as input.
+    """
+    if not lists:
+        return None
```
Comment (excellenteam-scaleup) [inline 5.2.cup_of_join.py:14]: Good explicit return

File: 5.1.thats_the_way.py (hunk 2 above)
Comment (excellenteam-scaleup) [reply 5.1.thats_the_way.py:21]: same goes to all other exercises

File: 5.3.parseltongue.py [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+
+    :param file_path: Path to the file containing the secret messages.
+    :type file_path: str
+    :return: A list of extracted secret messages.
+    :rtype: list[str]
+    """
+    secret_messages = []
+    current_message = ''
+
+    with open(file_path, 'rb') as file:
```
Comment (excellenteam-scaleup) [inline 5.3.parseltongue.py:14]: good use of open

File: 5.3.parseltongue.py [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+        while True:
```
Comment (excellenteam-scaleup) [inline 5.3.parseltongue.py:15]: You can do that with a more elegant way than while True.. you have an if statement

**Review by excellenteam-scaleup — COMMENTED** [review]
Good job overall.
Good documentation, convention, and testing in your main.

Take a look on the comments, and you fix them to get better score.
Try to implement the comments in your feature submissions.

**Pull request closed without merging**